uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -726,135 +757,112 @@ define([
-        vm.request.to_date_type = vm.request.from_date_type;
-      }
+      return $q.resolve()
+        .then(vm.setDatesFromUIExtended ? vm.setDatesFromUIExtended : _.noop)
+        .then(function () {
+          return checkAbsencePeriod(dateType);
+        });
+    }
+
+    /**
+     * Sets deductions in hours from UI to vm.request
+     */
+    function setHoursDeductionsToRequest () {
+      var times = vm.uiOptions.times;
+
+      vm.request.from_date_amount = !isNaN(+vm.uiOptions.times.from.amount) ? times.from.amount : null;
+      vm.request.to_date_amount = !isNaN(+vm.uiOptions.times.to.amount) ? times.to.amount : null;
     }
 
     /**
      * Sets the min and max for to date from absence period. It also sets the
      * init/starting date which user can select from. For multiple days request
      * user can select to date which is one more than the the start date.
      */
-    function _setMinMaxDate () {
+    function setMinMaxDatesToUI () {
       if (vm.uiOptions.fromDate) {
-        var nextFromDay = moment(vm.uiOptions.fromDate).add(1, 'd').toDate();
+        var nextFromDay = moment(vm.uiOptions.fromDate).add(1, 'day').toDate();
 
         vm.uiOptions.date.to.options.minDate = nextFromDay;
         vm.uiOptions.date.to.options.initDate = nextFromDay;
-
-        // also re-set to date if from date is changing and less than to date
-        if (vm.uiOptions.toDate && moment(vm.uiOptions.toDate).isBefore(vm.uiOptions.fromDate)) {
-          vm.uiOptions.toDate = vm.uiOptions.fromDate;
-        }
       } else {
-        vm.uiOptions.date.to.options.minDate = vm._convertDateFormatFromServer(vm.period.start_date);
+        vm.uiOptions.date.to.options.minDate = convertDateFormatFromServer(vm.period.start_date);
         vm.uiOptions.date.to.options.initDate = vm.uiOptions.date.to.options.minDate;
       }
 
-      vm.uiOptions.date.to.options.maxDate = vm._convertDateFormatFromServer(vm.period.end_date);
+      adjustDatesInUI();
+
+      vm.uiOptions.date.to.options.maxDate = convertDateFormatFromServer(vm.period.end_date);
     }
 
     /**
-     * Resets data in dates, types, balance.
-     */
-    function _reset () {
-      vm.uiOptions.toDate = vm.uiOptions.fromDate;
-      vm.request.to_date_type = vm.request.from_date_type;
-      vm.request.to_date = vm.request.from_date;
+     * Reset day types, times and deductions.
+     * Optionally show loading spinners.
+     *
+     * @param {String} dateType from|to
+     * @param {Boolean} showLoader
+     */
+    function resetAfterDateChange (dateType, showLoader) {
+      var time = vm.uiOptions.times[dateType];
+
+      vm['request' + _.startCase(dateType) + 'DayTypes'] = [];
+      time.time = '';
+      time.min = '00:00';
+      time.max = '00:00';
+      time.amount = 0;
+      time.maxAmount = 0;
+      time.disabled = true;
+
+      if (showLoader) {
+        time.loading = true;
+        vm.loading[dateType + 'DayTypes'] = true;
```

Comment:
we are putting the day types list (i assume) in "loading mode" but we're not doing anything afterwards with it, it seems the rest of the method is all about time. It's a bit confusing

Reply:
I agree, I simplified the func a little bit.